kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/KafkaRequestHandler.java

```diff
@@ -1241,40 +1314,73 @@ private void handleListOffsetRequestV0(KafkaHeaderAndRequest listOffset,
                                            CompletableFuture<AbstractResponse> resultFuture) {
         ListOffsetRequest request = (ListOffsetRequest) listOffset.getRequest();
 
-        Map<TopicPartition, CompletableFuture<ListOffsetResponse.PartitionData>> responseData = Maps.newHashMap();
+        List<CompletableFuture<Pair<TopicPartition, ListOffsetResponse.PartitionData>>>
+                allFutures = new ArrayList<>(request.offsetData().size());
```

Comment:
Why did you change the original `Map<TopicPartition, CompletableFuture<ListOffsetResponse.PartitionData>>` to `List<CompletableFuture<Pair<TopicPartition, ListOffsetResponse.PartitionData>>>`?

I mean, we can still use `Map` like:

```java
    private void handleListOffsetRequestV1AndAbove(KafkaHeaderAndRequest listOffset,
                                                   CompletableFuture<AbstractResponse> resultFuture) {
        ListOffsetRequest request = (ListOffsetRequest) listOffset.getRequest();

        final Map<TopicPartition, CompletableFuture<ListOffsetResponse.PartitionData>>
                partitionFutures = Maps.newConcurrentMap();

        request.partitionTimestamps().forEach(/* ... */);

        CompletableFuture.allOf(partitionFutures.values().toArray(new CompletableFuture[0]))
                .whenComplete((ignored, e) -> {
                    resultFuture.complete(new ListOffsetResponse(
                            CoreUtils.mapValue(partitionFutures, CompletableFuture::join)));
                });
    }
```

Reply:
This comment applies to both `handleListOffsetRequestV0` and `handleListOffsetRequestV1AndAbove`.